### ISSUE TITLE
Naming-Conventions: Allow _ AND format camelcase as parameter?

### ISSUE THREAD

**Issue by Stanja219** [typescript-eslint/typescript-eslint#1899]
**Repro**
In our project we have some places where we have an underscore as param.
`(_, args, { XXX}) `
Actually our params are supposed to be written in camelcase -format. BUT a single underscore is also valid. 
With the documentation I'm not able to find out if this case even can be configured with the availiable options.
I tried this, but this doesn't work. 

```TS
// your repro code case
  selector: 'parameter',
    format: ['camelCase'],
    filter: {
      regex: '/_/',
      match: true
    }
```

Can anybody help me?

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.26.0` |
| `@typescript-eslint/parser`        | `2.26.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.14.1` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
Does this option do what you want?

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md#leadingunderscore--trailingunderscore

**Comment by Stanja219** [comment]
Not really. When I allow underscores  _myParam would be also allowed, but actually we would like allow either _ OR myParam and forbid _myParam.

**Comment by bradzacher** [comment]
Oh sorry, I misunderstood. I'm with you now. That's a weird naming convention, but you do you!

I _believe_ you should be able to accomplish this with `custom`
```json5
{
  selector: 'parameter',
  format: null,
  custom: {
    regex: /(^_$|^[A-Za-z1-9$]+$)/,
    match: true
  }
}
```

`^_$` matches a single underscore
`^[A-Za-z1-9$]+$` (roughly) matches camelcase (you may have to add some additional characters if you're using "non-english" characters, etc.

**Comment by bbarry** [comment]
/^_$|^[a-z][A-Za-z1-9$]*$/

**Comment by Stanja219** [comment]
thanks for your help :-)
